### PR TITLE
Added build script for Elasticsearch 6.

### PR DESCRIPTION
**Pull request by markdorison**
Let me preface this by saying I have no idea what I am doing, but with this PR I was able to run `make elasticsearch-6` without an error. Beyond that, what should I be doing to verify this is working as expected?

### CODE REVIEW

**Comment by q0rban** [comment]
Nice work here!

At a quick glance, there seems to be a lot shared here with the elasticsearch-2.x image, so I wonder if there is a way of creating a baseimage for them without that baseimage showing up as an option in Tugboat?

**Comment by q0rban** [comment]
Ok, so after conferring with @bchavet, it looks like the `make push` command is where the magic happens for which images show up as options. So in theory we could move anything shared into a `elasticsearch-baseimage` image, and then do `FROM localhost:5000/elasticsearch-baseimage` in both this image and in the 2.4 image. I think that means we could keep the docker-entrypoint.sh script in the base image. (Assuming they are identical?)

**Comment by q0rban** [comment]
If we went this route, we would also need to add [dependencies](https://github.com/Lullabot/tugboat-registry/blob/master/Makefile#L8-L14) in the Makefile:

```
elasticsearch-2.4: elasticsearch-baseimage
elasticsearch-6: elasticsearch-baseimage
```

I guess my question for you, @markdorison, is do you think this is worth the effort, or are the images different enough that it they should be separate?

**Comment by markdorison** [comment]
@q0rban That makes sense to me in theory.

[This is the section](https://github.com/ChromaticHQ/tugboat-registry/blob/ba4266365111d6420e0bfceb4c9eb589a7f296bf/elasticsearch-6/Dockerfile#L79-L90) that is different in my 6.x image.

It seems clear to me how we could move the parts before it into a base image and pull in via the `FROM` command just as is done for the Ubuntu image, but what about the shared pieces that come after the install section? Can/should those be abstracted as well?

**Comment by q0rban** [comment]
Great question. I _think_ everything after that could also go in the baseimage _except_:

```
ENTRYPOINT ["/docker-entrypoint.sh"]
CMD ["elasticsearch"]
```

I think those will need to go in the "sub" images, otherwise I think the baseimage will fail to build because those commands won't work. This is a total guess on my part, not having tested this sort of thing before.

**Comment by q0rban** [comment]
The other thing that will probably fail is the chown elasticsearch:elasticsearch, since I doubt that user:group exists before elasticsearch is installed.

**Comment by q0rban** [comment]
It looks like the docker-entrypoint.sh handles the `chown`ing as well, so perhaps that could be removed from the baseimage.

**Comment by markdorison** [comment]
Updated with `elasticsearch-baseimage`.

File: Makefile [hunk 1]
```diff
@@ -12,6 +12,8 @@ apache-php7: apache
 apache-php7-drupal: apache-php7
 nginx-php: nginx
 nginx-php7: nginx
+elasticsearch-2.4: elasticsearch-baseimage
+elasticsearch-6: elasticsearch-baseimage
```
Comment (q0rban) [inline Makefile:16]: 👏

File: Makefile [hunk 2]
```diff
@@ -34,4 +36,4 @@ tugboat-init:
 	rm -rf /var/www/__tugboat
 
 tugboat-build:
-	a2enmod include
\ No newline at end of file
+	a2enmod include
```
Comment (q0rban) [inline Makefile:39]: :bowtie:

**Review by q0rban — CHANGES_REQUESTED** [review]
I can't get the elasticsearch-6 container to work because the docker-entrypoint.sh expects gosu to be present.

```
$ make elasticsearch-6
...
$ docker create localhost:5000/elasticsearch-6:latest elasticsearch
a8defe741209723dc1bb495a0900a718bbe047cc8ff69bcb7810db9977f02baf
$ docker start a8defe74 -i
/docker-entrypoint.sh: line 28: exec: gosu: not found
```

It works with elasticsearch-2.4:

```
$ make elasticsearch-2.4
...
$ docker create localhost:5000/elasticsearch-2.4:latest elasticsearch
b4b0c957585daa0b43cc202912d562c6a449d95e68a7d0437cb6c855c39b16c1
$ docker start b4b0c957 -i
[2018-03-05 17:43:24,610][WARN ][bootstrap                ] unable to install syscall filter: seccomp unavailable: your kernel is buggy and you should upgrade
[2018-03-05 17:43:24,741][INFO ][node                     ] [Jebediah Guthrie] version[2.4.6], pid[1], build[5376dca/2017-07-18T12:17:44Z]
[2018-03-05 17:43:24,742][INFO ][node                     ] [Jebediah Guthrie] initializing ...
[2018-03-05 17:43:25,201][INFO ][plugins                  ] [Jebediah Guthrie] modules [reindex, lang-expression, lang-groovy], plugins [], sites []
[2018-03-05 17:43:25,251][INFO ][env                      ] [Jebediah Guthrie] using [1] data paths, mounts [[/ (overlay)]], net usable_space [49.3gb], net total_space [62.7gb], spins? [possibly], types [overlay]
[2018-03-05 17:43:25,251][INFO ][env                      ] [Jebediah Guthrie] heap size [990.7mb], compressed ordinary object pointers [true]
[2018-03-05 17:43:26,593][INFO ][node                     ] [Jebediah Guthrie] initialized
[2018-03-05 17:43:26,594][INFO ][node                     ] [Jebediah Guthrie] starting ...
[2018-03-05 17:43:26,650][INFO ][transport                ] [Jebediah Guthrie] publish_address {172.17.0.5:9300}, bound_addresses {0.0.0.0:9300}
[2018-03-05 17:43:26,655][INFO ][discovery                ] [Jebediah Guthrie] elasticsearch/nKKcxNAMQlSNJoIKq8Jo0w
[2018-03-05 17:43:29,719][INFO ][cluster.service          ] [Jebediah Guthrie] new_master {Jebediah Guthrie}{nKKcxNAMQlSNJoIKq8Jo0w}{172.17.0.5}{172.17.0.5:9300}, reason: zen-disco-join(elected_as_master, [0] joins received)
[2018-03-05 17:43:29,748][INFO ][http                     ] [Jebediah Guthrie] publish_address {172.17.0.5:9200}, bound_addresses {0.0.0.0:9200}
[2018-03-05 17:43:29,748][INFO ][node                     ] [Jebediah Guthrie] started
[2018-03-05 17:43:29,768][INFO ][gateway                  ] [Jebediah Guthrie] recovered [0] indices into cluster_state
```

**Comment by q0rban** [comment]
Some digging around:

- https://github.com/elastic/elasticsearch-docker/blob/master/templates/Dockerfile.j2
- https://github.com/elastic/elasticsearch-docker/blob/6.2/build/elasticsearch/bin/docker-entrypoint.sh

Maybe these images aren't so similar. It's too bad we can't just use official images directly, because this one is way over my head.

**Comment by markdorison** [comment]
> I can't get the elasticsearch-6 container to work because the docker-entrypoint.sh expects gosu to be present.

I am able to reproduce your issue. I was _not_ previously using those steps to test. The steps I was following were:

1. `make elasticsearch-6`
1. `make push elasticsearch-6`
1. `docker run -i -t localhost:5000/elasticsearch-6 /bin/bash`

When I run step 3 I get a bash prompt from the container with no error. Do you know what the difference is between what I was doing vs what you did? Maybe the official steps to test can be added to the README?

> Maybe these images aren't so similar. It's too bad we can't just use official images directly, because this one is way over my head.

If this is over your head, I am probably drowning. Maybe I should pause until we know if this is the right direction.

**Comment by q0rban** [comment]
> Maybe I should pause until we know if this is the right direction.

Related: #24

I wonder if this technique could work for ElasticSearch 6: https://github.com/moby/moby/issues/3465#issuecomment-313389014

**Comment by q0rban** [comment]
Also, 👍 to this:

> Maybe the official steps to test can be added to the README?

I am not sure myself, so I usually just try a few things until something works. 😬

**Comment by markdorison** [comment]
I believe I have solved the `gosu` issue. I [moved `gosu` installation to the baseimage](https://github.com/Lullabot/tugboat-registry/pull/23/commits/585800a60c64dfbc5062076a3f50b4ac6443b61e) and ran the following to test:

```
$ docker create localhost:5000/elasticsearch-6:latest elasticsearch
b01a97824ad6c1c0e4f3985fdc45c39ccb2493a959321a9db88bb497abd70574
$ docker start b01a97824ad6c1c0e4f3985fdc45c39ccb2493a959321a9db88bb497abd70574 -i
[2018-03-05T18:54:20,261][INFO ][o.e.n.Node               ] [] initializing ...
[2018-03-05T18:54:20,446][INFO ][o.e.e.NodeEnvironment    ] [SmY7yOm] using [1] data paths, mounts [[/ (overlay)]], net usable_space [54.7gb], net total_space [62.7gb], types [overlay]
[2018-03-05T18:54:20,447][INFO ][o.e.e.NodeEnvironment    ] [SmY7yOm] heap size [1007.3mb], compressed ordinary object pointers [true]
[2018-03-05T18:54:20,449][INFO ][o.e.n.Node               ] node name [SmY7yOm] derived from node ID [SmY7yOmbTAKx9Xb-xLkbqg]; set [node.name] to override
[2018-03-05T18:54:20,450][INFO ][o.e.n.Node               ] version[6.2.2], pid[1], build[10b1edd/2018-02-16T19:01:30.685723Z], OS[Linux/4.9.60-linuxkit-aufs/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_141/25.141-b15]
[2018-03-05T18:54:20,451][INFO ][o.e.n.Node               ] JVM arguments [-Xms1g, -Xmx1g, -XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+UseCMSInitiatingOccupancyOnly, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.io.tmpdir=/tmp/elasticsearch.aR9DQls8, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=/var/lib/elasticsearch, -XX:+PrintGCDetails, -XX:+PrintGCDateStamps, -XX:+PrintTenuringDistribution, -XX:+PrintGCApplicationStoppedTime, -Xloggc:/var/log/elasticsearch/gc.log, -XX:+UseGCLogFileRotation, -XX:NumberOfGCLogFiles=32, -XX:GCLogFileSize=64m, -Des.path.home=/usr/share/elasticsearch, -Des.path.conf=/etc/elasticsearch]
[2018-03-05T18:54:21,434][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [aggs-matrix-stats]
[2018-03-05T18:54:21,435][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [analysis-common]
[2018-03-05T18:54:21,436][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [ingest-common]
[2018-03-05T18:54:21,436][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [lang-expression]
[2018-03-05T18:54:21,437][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [lang-mustache]
[2018-03-05T18:54:21,437][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [lang-painless]
[2018-03-05T18:54:21,438][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [mapper-extras]
[2018-03-05T18:54:21,439][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [parent-join]
[2018-03-05T18:54:21,440][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [percolator]
[2018-03-05T18:54:21,440][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [rank-eval]
[2018-03-05T18:54:21,441][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [reindex]
[2018-03-05T18:54:21,441][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [repository-url]
[2018-03-05T18:54:21,442][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [transport-netty4]
[2018-03-05T18:54:21,442][INFO ][o.e.p.PluginsService     ] [SmY7yOm] loaded module [tribe]
[2018-03-05T18:54:21,443][INFO ][o.e.p.PluginsService     ] [SmY7yOm] no plugins loaded
[2018-03-05T18:54:25,582][INFO ][o.e.d.DiscoveryModule    ] [SmY7yOm] using discovery type [zen]
[2018-03-05T18:54:26,535][INFO ][o.e.n.Node               ] initialized
[2018-03-05T18:54:26,536][INFO ][o.e.n.Node               ] [SmY7yOm] starting ...
[2018-03-05T18:54:26,939][INFO ][o.e.t.TransportService   ] [SmY7yOm] publish_address {127.0.0.1:9300}, bound_addresses {127.0.0.1:9300}
[2018-03-05T18:54:30,137][INFO ][o.e.c.s.MasterService    ] [SmY7yOm] zen-disco-elected-as-master ([0] nodes joined), reason: new_master {SmY7yOm}{SmY7yOmbTAKx9Xb-xLkbqg}{YWR1RAqHS0moxMb3HbJapQ}{127.0.0.1}{127.0.0.1:9300}
[2018-03-05T18:54:30,153][INFO ][o.e.c.s.ClusterApplierService] [SmY7yOm] new_master {SmY7yOm}{SmY7yOmbTAKx9Xb-xLkbqg}{YWR1RAqHS0moxMb3HbJapQ}{127.0.0.1}{127.0.0.1:9300}, reason: apply cluster state (from master [master {SmY7yOm}{SmY7yOmbTAKx9Xb-xLkbqg}{YWR1RAqHS0moxMb3HbJapQ}{127.0.0.1}{127.0.0.1:9300} committed version [1] source [zen-disco-elected-as-master ([0] nodes joined)]])
[2018-03-05T18:54:30,239][INFO ][o.e.h.n.Netty4HttpServerTransport] [SmY7yOm] publish_address {127.0.0.1:9200}, bound_addresses {127.0.0.1:9200}
[2018-03-05T18:54:30,239][INFO ][o.e.n.Node               ] [SmY7yOm] started
[2018-03-05T18:54:30,265][INFO ][o.e.g.GatewayService     ] [SmY7yOm] recovered [0] indices into cluster_state
```

**Review by q0rban — APPROVED** [review]
Works for me! I'll wait for @bchavet to review and merge. Nice work!

@markdorison++

**Pull request merged**